src/components/ContactsList.js

```diff
@@ -1,31 +1,39 @@
-import { useState } from "react"
-import { Link, useSearchParams } from "react-router-dom"
+import { Link } from "react-router-dom"
 
-function ContactsList(props) {
-  
-  //"contacts" must be passed as prop to this component
-  const { contacts } = props
+function ContactsList (props) {
+  const { contacts, setType, contactType } = props
 
   return (
     <>
       <header>
         <h2>Contacts</h2>
       </header>
+      <select
+        id="type"
+        value={ contactType }
+        onChange={ e => setType(e.target.value) }
+        name="type"
+      >
+        <option value="">--No filter selected--</option>
```

Comment:
Not sure if you implemented the `useSearchParam` with this. Which changes your URL for easier navigation and structure.